lib/appnexusapi/services/report_service.rb

```diff
@@ -0,0 +1,48 @@
+module AppnexusApi
+  class ReportService < AppnexusApi::Service
+    REPORT_READY = 'ready'.freeze
+
+    def initialize(connection)
+      @read_only = false
+      super(connection)
+    end
+
+    def report_meta_data(report_name)
+      fail('Report name has to be defined!') if report_name.nil? || report_name.empty?
+
+      meta_data = get('meta' => report_name)
+      return nil if meta_data.nil?
+
+      meta_data.first.raw_json
```

Comment:
are there any cases where `first` isn't what we want? is `get` actually taking a partial match, instead of an exact report name match?

Reply:
`get` always returns an array, but this get has a single response, a hash that describes the possible config params for the report, so we always want the `first`. if the report name is not complete than i expect the api to return an error, but i didn't tried yet, will give it a try